synapse/events/utils.py

```diff
@@ -518,15 +529,21 @@ def _inject_bundled_aggregations(
         if event_aggregations.replace:
             # If there is an edit, optionally apply it to the event.
             edit = event_aggregations.replace
-            if apply_edits:
+            if apply_edits and not self._inhibit_replacement_via_edits:
                 self._apply_edit(event, serialized_event, edit)
 
             # Include information about it in the relations dict.
-            serialized_aggregations[RelationTypes.REPLACE] = {
-                "event_id": edit.event_id,
-                "origin_server_ts": edit.origin_server_ts,
-                "sender": edit.sender,
-            }
+            #
+            # Matrix spec v1.5 (https://spec.matrix.org/v1.5/client-server-api/#server-side-aggregation-of-mreplace-relationships)
+            # said that we should only include the `event_id`, `origin_server_ts` and
+            # `sender` of the edit; however MSC3925 proposes extending it to the whole
+            # of the edit, which is what we do here.
+            serialized_aggregations[RelationTypes.REPLACE] = self.serialize_event(
+                edit,
+                time_now,
+                config=config,
```

Comment:
I think this is slightly different than MSC3925 which says to include "the entire content of the most recent replacement event, formatted as described in [Room Event Format](https://spec.matrix.org/v1.4/client-server-api/#room-event-format)."

The code matches what we do for threaded events and is generally what makes more sense to me (matching the serialization of other events in the response).

Reply:
> I think this is slightly different than [MSC3925](https://github.com/matrix-org/matrix-spec-proposals/pull/3925) which says to include "the entire content of the most recent replacement event, formatted as described in [Room Event Format](https://spec.matrix.org/v1.4/client-server-api/#room-event-format)."

I'm not really following. How does this differ from the proposal?

Reply:
Ah, the spec link confused me -- that's kind of just describing the entire event format, unrelated to how it might be modified by a filter.

Ignore me!